grammars/ruby.cson

```diff
@@ -1860,7 +1860,7 @@
     ]
   }
   {
-    'begin': '(?<={|do|{\\s|do\\s)(\\|)'
+    'begin': '(?<={|{\\s|[^A-Za-z0-9_]do|^do|[^A-Za-z0-9_]do\\s|^do\\s)(\\|)'
```

Comment:
This is absolutely horrible, but it seems like the best solution without a large rewrite.  Thanks for the PR.